src/enqueue_links/click_elements.js

```diff
@@ -0,0 +1,425 @@
+ *   which lets you specify special properties for the enqueued {@link Request} objects.
+ *
+ *   If `pseudoUrls` is an empty array, `null` or `undefined`, then the function
+ *   enqueues all links found on the page.
+ * @param {Function} [options.updateRequestFunction]
+ *   Just before a new {@link Request} is enqueued to the {@link RequestQueue}, this function can be used to modify
+ *   its contents such as `userData`, `payload` or, most importantly `uniqueKey`. This is useful when necessary
+ *   to enqueue multiple `Requests` to the queue that share the same URL, but differ in methods or payloads.
+ *
+ *   For example: by generating your own `uniqueKey` from a combination of `url`, `method` and `payload` you enable crawling
+ *   of websites that navigate using form submits (POST requests).
+ * @param {number} [options.waitForPageIdleSecs=1]
+ *   Clicking in the page triggers various asynchronous operations that lead to new URLs being shown
+ *   by the browser. It could be a simple JavaScript redirect or opening of a new tab in the browser.
+ *   These events often happen only some time after the actual click. Requests typically take milliseconds
+ *   while new tabs open in hundreds of milliseconds.
+ *
+ *   To be able to capture all those events, the `enqueueLinksByClickingElements()` function repeatedly waits
+ *   for the `waitForPageIdleSecs`. By repeatedly we mean that whenever a relevant event is triggered, the timer
+ *   is restarted. As long as new events keep coming, the function will not return, unless
+ *   the below `maxWaitForPageIdleSecs` timeout is reached.
+ *
+ *   You may want to reduce this for example when you're sure that your clicks do not open new tabs,
+ *   or increase when you're not getting all the expected URLs.
+ * @param {number} [options.maxWaitForPageIdleSecs=5]
+ *   This is the maximum period for which the function will keep tracking events, even if more events keep coming.
+ *   Its purpose is to prevent a deadlock in the page by periodic events, often unrelated to the clicking itself.
+ *   See `waitForPageIdleSecs` above for an explanation.
+ * @return {Promise<QueueOperationInfo[]>}
+ *   Promise that resolves to an array of {@link QueueOperationInfo} objects.
+ * @memberOf puppeteer
+ * @name enqueueLinksByClickingElements
+ */
+export async function enqueueLinksByClickingElements(options = {}) {
+    const {
+        page,
+        requestQueue,
+        selector,
+        pseudoUrls,
+        updateRequestFunction,
+        waitForPageIdleSecs = 1,
+        maxWaitForPageIdleSecs = 5,
+    } = options;
+
+    checkParamOrThrow(page, 'page', 'Object');
+    checkParamOrThrow(selector, 'selector', 'String');
+    checkParamPrototypeOrThrow(requestQueue, 'requestQueue', [RequestQueue, RequestQueueLocal], 'Apify.RequestQueue');
+    checkParamOrThrow(pseudoUrls, 'pseudoUrls', 'Maybe Array');
+    checkParamOrThrow(updateRequestFunction, 'updateRequestFunction', 'Function');
```

Comment:
I would consider name tranformRequestFunction as it doesn't update requests in place but transforms them. Also I would add `.filter(request => !!request)` so that you can remove some requests in tranformation.

Reply:
Good idea!